infrastructure/src/test/java/org/corfudb/infrastructure/BaseServerTest.java

```diff
@@ -182,37 +177,6 @@ public void testSealWrongEpoch() {
         assertEquals(2L, response.getPayload().getServerError().getWrongEpochError().getCorrectEpoch());
     }
 
-    /**
-     * Test that the BaseServer correctly handles a VERSION_REQUEST.
-     */
-    @Test
-    public void testVersionRequest() throws SerializerException {
-        RequestMsg request = getRequestMsg(
-                getBasicHeader(ClusterIdCheck.IGNORE, EpochCheck.IGNORE),
-                getVersionRequestMsg()
-        );
-
-        ArgumentCaptor<ResponseMsg> responseCaptor = ArgumentCaptor.forClass(ResponseMsg.class);
-        when(mockServerContext.getServerConfig()).thenReturn(Collections.singletonMap("--address", "localhost"));
-        when(mockServerContext.getNodeIdBase64()).thenReturn("localhost");
-
-        baseServer.handleMessage(request, mockChannelHandlerContext, mockServerRouter);
-
-        verify(mockServerRouter).sendResponse(responseCaptor.capture(), any(ChannelHandlerContext.class));
-        ResponseMsg response = responseCaptor.getValue();
-
-        // Assert that the payload has a VERSION_RESPONSE and that the base
-        // header fields have remained the same
-        assertTrue(compareBaseHeaderFields(request.getHeader(), response.getHeader()));
-        assertTrue(response.getPayload().hasVersionResponse());
-
-        // Assert that the contents of the versionInfo are what we expect
-        VersionInfo versionInfo = getVersionInfo(response.getPayload().getVersionResponse());
-        assertEquals("localhost", versionInfo.getNodeId());
-        assertEquals(1, versionInfo.getOptionsMap().size());
-        assertEquals("localhost", versionInfo.getOptionsMap().get("--address"));
-    }
-
```

Comment:
If required, can we write some test cases to check that the warning message was printed when there is a version mismatch and was not printed when there is no mismatch? Later this can be extended to checking the thrown exceptions or printing some other error/warn messages.

Reply:
Unit tests added, thanks!